tests/AddressRegistry.spec.ts

```
@@ -0,0 +1,122 @@
+      // add new entry
+      await addressRegistry.addAddress(accounts[2].address);
+      expect(await addressRegistry.length()).toEqBN("2");
+    });
+
+    it("get correct index", async () => {
+      expect(await addressRegistry.getIndex(accounts[1].address)).toEqBN("0");
+      expect(await addressRegistry.getIndex(accounts[2].address)).toEqBN("1");
+
+      await expect(addressRegistry.getIndex(accounts[4].address)).toBeRevertedWith(
+        "EnumerableSet: Index not found"
+      );
+    });
+
+    it("can list all values of the registry", async () => {
+      let length = await (await addressRegistry.length()).toNumber();
+      let values = [];
+
+      for (let i = 0; i < length; i++) {
+        values.push(await addressRegistry.getAddress(i));
+      }
+
+      expect(values).toStrictEqual([accounts[1].address, accounts[2].address]);
+    });
+
+    it("can get all raw values in one call", async () => {
+      expect(await addressRegistry.getAllRawValues()).toStrictEqual([
+        "0x000000000000000000000000" + accounts[1].address.replace("0x", "").toLowerCase(),
+        "0x000000000000000000000000" + accounts[2].address.replace("0x", "").toLowerCase(),
+      ]);
+    });
+  });
+
+  // Require ERC725
+  describe("AddressRegistryRequiresERC725", () => {
+    let addressRegistryRequireERC725: AddressRegistryRequiresERC725,
+      account: LSP3Account,
+      owner: SignerWithAddress;
+
+    beforeEach(async () => {
+      owner = accounts[3];
+      account = await new LSP3Account__factory(owner).deploy(owner.address);
+      addressRegistryRequireERC725 = await new AddressRegistryRequiresERC725__factory(
+        owner
+      ).deploy();
+    });
+
+    it("add address", async () => {
+      let abi = addressRegistryRequireERC725.interface.encodeFunctionData("addAddress", [
+        account.address,
+      ]);
+
+      await account.execute(0, addressRegistryRequireERC725.address, 0, abi, {
+        from: owner.address,
+        gasLimit: 3_000_000,
```

Not specifying this option make the contract call fail (with no state update / balance change at the target contract / account).
Could be a bug in the low level call of ERC725X.

For:
- **web3**: fails with `revert`
- **etherjs**: fails silently